# HG changeset patch
# User Rik <rik@octave.org>
# Date 1546454079 28800
#      Wed Jan 02 10:34:39 2019 -0800
# Branch stable
# Node ID 190702f4a075d98800cc22d0ef4586301f49b214
# Parent  7a6f7a81ccd041267ddb0bc07a83d418ca7a205d
mkoctfile.in.cc: Fix static analyzer detected issues (bug #55347).

* mkoctfile.in.cc (do_getenv): Delete function.
* mkoctfile.in.cc (get_temp_directory): Replace calls to do_getenv with calls
to octave_getenv from shared-fcns.h.

diff --git a/src/mkoctfile.in.cc b/src/mkoctfile.in.cc
--- a/src/mkoctfile.in.cc
+++ b/src/mkoctfile.in.cc
@@ -549,48 +549,40 @@ run_command (const std::string& cmd, boo
 
 bool
 is_true (const std::string& s)
 {
   return (s == "yes" || s == "true");
 }
 
 static std::string
-do_getenv (const std::string& name)
-{
-  char *value = ::getenv (name.c_str ());
-
-  return value ? value : "";
-}
-
-static std::string
 get_temp_directory (void)
 {
   std::string tempd;
 
 #if defined (__MINGW32__) || defined (_MSC_VER)
 
-  tempd = do_getenv ("TEMP");
+  tempd = octave_getenv ("TEMP");
 
   if (tempd.empty ())
-    tempd = do_getenv ("TMP");
+    tempd = octave_getenv ("TMP");
 
 #if defined (P_tmpdir)
   if (tempd.empty ())
     tempd = P_tmpdir;
 #endif
 
   // Some versions of MinGW and MSVC either don't define P_tmpdir, or
   // define it to a single backslash.  In such cases just use C:\temp.
   if (tempd.empty () || tempd == R"(\)")
     tempd = R"(c:\temp)";
 
 #else
 
-  tempd = do_getenv ("TMP");
+  tempd = octave_getenv ("TMP");
 
 #if defined (P_tmpdir)
   if (tempd.empty ())
     tempd = P_tmpdir;
 #else
   if (tempd.empty ())
     tempd = "/tmp";
 #endif
